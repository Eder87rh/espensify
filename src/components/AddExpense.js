import React from 'react'

export default () => (
    <div>
        This is from my add expense Component!!
    </div>
)