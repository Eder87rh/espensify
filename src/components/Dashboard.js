import React from 'react'

export default () => (
    <div>
        This is from my dashboard Component
    </div>
)