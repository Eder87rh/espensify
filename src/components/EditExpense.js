import React from 'react'

export default (props) => {
    console.log(props)
    return (
        <div>
            Editing the espense with id of { props.match.params.id }
        </div>
    )
}