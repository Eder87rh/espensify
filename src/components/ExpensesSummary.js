import React from 'react';
import { connect } from 'react-redux';
import { getTotalExpenser } from '../selectors/expenses-total'
import expensesFilter from '../selectors/expenses'
import numeral from 'numeral'

export const ExpensesSummary = ({ count, total }) => {
  console.log("TCL: ExpensesSummary -> total", total)
  const expenseWord = count === 1 ? 'expense' : 'expenses'; 
  const totalFormated = numeral(total /100).format('$0,0.00');

  return (
    <h1>
      Viewing { count } { expenseWord } totalling { totalFormated }
    </h1>
  );
}

const mapStateToProps = state => {
  const visibleExpenses = expensesFilter(state.expenses, state.filters)

  return {
    count: visibleExpenses.length,
    total: getTotalExpenser(expensesFilter(state.expenses, state.filters))
  }
}


 
export default connect(mapStateToProps)(ExpensesSummary);