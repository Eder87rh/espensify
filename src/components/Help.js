import React from 'react'

export default () => (<div>This is from my help expense PAGE!!</div>)