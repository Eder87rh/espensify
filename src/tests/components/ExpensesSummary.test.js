import React from 'react';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import { shallow } from 'enzyme'
import expenses from '../fixtures/expenses'

/* test('Should render correctly without expenses', () => {
  const wrapper = shallow(<ExpensesSummary  count={0} total={0}/>)
  const text = wrapper.find('h1').text()
  
  expect(wrapper).toMatchSnapshot(wrapper)
  expect(text).toBe('Viewing 0 expenses totalling $0.00')
}) */

test('Should render correctly with expenses', () => {
  const wrapper = shallow(<ExpensesSummary count={3} total={10000}/>)
  const text = wrapper.find('h1').text()
  expect(wrapper).toMatchSnapshot(wrapper)
  expect(text).toBe('Viewing 3 expenses totalling $100.00')
})