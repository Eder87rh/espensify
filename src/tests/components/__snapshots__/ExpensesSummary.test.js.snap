// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[object Object] 1`] = `
<h1>
  Viewing 
  3
   
  expenses
   totalling 
  $100.00
</h1>
`;
